02-peer-protocol.md

```diff
@@ -606,13 +619,22 @@ A receiving node which closes an incoming HTLC in response to an
 `update-fail-htlc` message on an offered HTLC MUST copy the `reason`
 field to the outgoing `update-fail-htlc`.
 
+A receiving node MUST fail the channel if the `BADONION` bit in
+`failure-code` is not set for `update-fail-malformed-htlc`.
 
-#### Rationale
+A receiving node which closes an incoming HTLC in response to an
```

Comment:
Yeah, wording is awkward, I prefer yours.

It currently copies the entire reason field (which is failure-code + sha256-of-onion + anything we add later).  I think that's clearer for the bitflip case, but...

Perhaps we should tighten the spec, and say all BADONION messages MUST have contain only the sha256-of-onion as `reason`, and the node receiving update-fail-malformed-htlc MAY choose a different failure (or retransmission) if it does not match the onion it sent?

That allows you do detect bitflips from C and choose what to do.  I expect most will simply punt upstream.